docs/source/essentials/normalized-cache.mdx

```diff
@@ -0,0 +1,350 @@
+    // Retrieve the id from the field arguments.
+    // In the example, this allows to know that `author(id: "author1")` will retrive `author1`
+    // That sounds straightforward but without this, the cache would have no way of finding the id before executing the request on the
+    // network which is what we want to avoid
+    return CacheKey.from(field.resolveArgument("id", variables) as String)
+  }
+}
+
+val apolloClient = ApolloClient.builder()
+    .serverUrl("https://...")
+    .normalizedCache(cacheFactory, resolver)
+    .build()
+```
+
+```java:title=Java
+CacheKeyResolver resolver = new CacheKeyResolver() {
+   @NotNull @Override
+   public CacheKey fromFieldRecordSet(@NotNull ResponseField field, @NotNull Map<String, Object> recordSet) {
+     // Retrieve the id from the object itself
+     return CacheKey.from(((String) recordSet.get("id"));
+   }
+
+   @NotNull @Override
+   public CacheKey fromFieldArguments(@NotNull ResponseField field, @NotNull Operation.Variables variables) {
+     // Retrieve the id from the field arguments.
+     // In the example, this allows to know that `author(id: "author1")` will retrive `author1`
+     // That sounds straightforward but without this, the cache would have no way of finding the id before executing the request on the
+     // network which is what we want to avoid
+     return CacheKey.from(((String) field.resolveArgument("id", variables));
+   }
+};
+
+//Build the ApolloClient
+ApolloClient apolloClient = ApolloClient.builder()
+  .serverUrl("https://...")
+  .normalizedCache(cacheFactory, resolver)
+  .build();
+```
+
+</MultiCodeBlock>
+
+For this resolver to work, every object in your graph needs to have a globally unique ID. If some of them don't have one, you can fallback to using the path as cache key by returning `CacheKey.NO_KEY`.
+
+## Using the cache with your queries
+
+You control how the cache is used with [`ResponseFetchers`](https://github.com/apollographql/apollo-android/blob/master/apollo-runtime/src/main/java/com/apollographql/apollo/fetcher/ApolloResponseFetchers.java):
+
+<MultiCodeBlock>
+
+```kotlin
+// Get a response from the cache if possible. Else, get a response from the network
+// This is the default behavior
+val apolloCall = apolloClient().query(BookWithAuthorName()).responseFetcher(ApolloResponseFetchers.CACHE_FIRST)
+```
+
+```java
+// Get a response from the cache if possible. Else, get a response from the network
+// This is the default behavior
+ApolloCall apolloCall = apolloClient().query(new BookWithAuthorName()).responseFetcher(ApolloResponseFetchers.CACHE_FIRST)
+```
+
+</MultiCodeBlock>
+
+Other possibilities are CACHE_ONLY, NETWORK_ONLY, CACHE_AND_NETWORK_ONLY, NETWORK_FIST. See to the [`ResponseFetchers` class](https://github.com/apollographql/apollo-android/blob/master/apollo-runtime/src/main/java/com/apollographql/apollo/fetcher/ApolloResponseFetchers.java) for more details.
```

Comment:
`code voice` for each constant

add 'and' between the last two constant values